### PR TITLE
Add Upnp Init retry logic.

### PR DESCRIPTION
**Pull request by whyman**
Tidy up startup and error messaging.
Log IPv6 Addresses/ports too.
Add warning if user is providing an IP instead of interface.

### CODE REVIEW

File: src/server.cc [hunk 1]
```diff
@@ -34,6 +34,7 @@
 #ifdef HAVE_CURL
 #include <curl/curl.h>
 #include <iostream>
+#include <upnptools.h>
```
Comment (KarlStraussberger) [inline src/server.cc:37]: This should go outside #ifdef

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]

File: src/server.cc (hunk 1 above)
Comment (whyman) [reply src/server.cc:41]: Yep, it needs its own #ifdef! Not sure why editor decided to add it there :grin:

**Comment by whyman** [comment]
Hmmm, Conan has not got `1.14.x` yet :-1:

**Comment by whyman** [comment]
Conan PR: https://github.com/conan-io/conan-center-index/pull/5008

Currently blocked with a libtool error on `macos` and a question about security.

**Comment by phi-whiterabbit** [comment]
Fwiw: I cannot compile this branch with -DWITH_NPUPNP.

**Comment by whyman** [comment]
Conan PR was updated, hopefully it passes the CI this time...

**Review by KarlStraussberger — APPROVED** [review]

**Comment by whyman** [comment]
Yay the conan PR is finally merged!

**Pull request merged**